### PR TITLE
InfoViz fixes

### PR DESCRIPTION
**Pull request by scottwittenburg**

### CODE REVIEW

**Comment by scottwittenburg** [comment]
@aronhelser Mind taking a look at the histogram selector changes?

File: src/InfoViz/Native/HistogramSelector/score.js [hunk 1]
```diff
@@ -126,7 +126,13 @@ export default function init(inPublicAPI, inModel) {
...
         scoreData.name = `${scoreData.selection.partition.variable} (partition)`;
       }
-      model.provider.setAnnotation(scoreData);
+      if (!passive) {
+        model.provider.setAnnotation(scoreData);
+      } else if (model.provider.isA('AnnotationStoreProvider') && model.provider.getStoredAnnotation(scoreData.id)) {
+        model.provider.updateStoredAnnotations({
+          [scoreData.id]: scoreData,
+        });
+      }
```
Comment (aronhelser) [inline src/InfoViz/Native/HistogramSelector/score.js]: Not passive means this change triggers it becoming the active annotation. 
But a passive update still will update the stored annotation, without it becoming active? If I'm reading it correctly, I think it makes sense, but add a comment to the passive case?

File: src/InfoViz/Native/HistogramSelector/score.js [hunk 2]
```diff
@@ -859,20 +865,22 @@ export default function init(inPublicAPI, inModel) {
...
-            divider.value = (((index + 1) / (def.dividers.length + 1)) *
-                             (hobj.max - hobj.min)) + hobj.min;
-          } else {
-            // this set the divider to hobj.min if the new hobj.min === hobj.max.
-            divider.value = (((divider.value - oldRangeMin) / (oldRangeMax - oldRangeMin)) *
-                             (hobj.max - hobj.min)) + hobj.min;
-          }
-        });
-        sendScores(def);
+      if (showScore(def)) {
```
Comment (aronhelser) [inline src/InfoViz/Native/HistogramSelector/score.js]: I think we put in this guard while working together, but now I don't remember why we needed it - add a comment?

File: src/InfoViz/Native/HistogramSelector/index.js [hunk 3]
```diff
@@ -784,6 +782,13 @@ function histogramSelector(publicAPI, model) {
     });
   }
 
+  function createFieldData(fieldName) {
+    return Object.assign(
+      model.fieldData[fieldName] || {},
+      model.provider.getField(fieldName),
+      scoreHelper.defaultFieldData());
```
Comment (jourdain) [inline src/InfoViz/Native/HistogramSelector/index.js]: Are the default meant to override any of the possible keys?

Comment (aronhelser) [inline src/InfoViz/Native/HistogramSelector/index.js]: No, they aren't. For future-proofing, when I think about swapping the order, I'm not sure the fieldProvider would have any information that should override the scoreHelper, though. Think it's OK to leave it?

Comment (jourdain) [inline src/InfoViz/Native/HistogramSelector/index.js]: ok, just wanted to mention it while I was reviewing it.

**Pull request merged**